File: C#/StreamView/StreamView/View/IByteStreamView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StreamView.View
{
    interface IByteStreamView
    {
        #region Methods

        /// <summary>
        /// Sets the list of bytes that are displayed in this view.
        /// </summary>
        void setBytes( Byte[] bytes );

        /// <summary>
        /// Sets the list of bytes that are displayed in this view. The
        /// "offset" parameter can be used to indicate the offset in the
        /// underlying data where these bytes belong.
        /// </summary>
        /// <param name="bytes"></param>
        /// <param name="offset"></param>
        void setBytes( Byte[] bytes, long offset );

        /// <summary>
        /// Highlight one or more bits (and their corresponding bytes)
        /// in this view.
        /// </summary>
        /// <param name="bitOffset"></param>
        /// <param name="bitLength"></param>
        void highlightSelection( int bitOffset, int bitLength );

        #endregion
    }
}

File: Java/StreamView/src/view/IByteStreamView.java
package view;

/*******************************************************************************
 * Interface for showing and highlighting some bytes.
 ******************************************************************************/
public interface IByteStreamView
{
    /***************************************************************************
     * Sets the bytes to be displayed in this view.
     * 
     * @param bytes
     **************************************************************************/
    void setBytes( byte[] bytes );

    /***************************************************************************
     * Sets the bytes, and their offset in some stream, to be displayed in this
     * view.
     * 
     * @param bytes
     * @param offset
     **************************************************************************/
    void setBytes( byte[] bytes, long offset );

    /***************************************************************************
     * Highlights some bits in this view.
     * 
     * @param bitOffset
     * @param bitLength
     **************************************************************************/
    void highlightSelection( int bitOffset, int bitLength );
}
